### ISSUE TITLE
Improve diagnostic when string from property is not assignable to string literal.

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/TypeScript#26413]
**TypeScript Version:**  3.1.0-dev.20180810

**Code**

```ts
interface IceCreamOptions {
    readonly cherry: "yes" | "no"
}
declare function iceCream(options: IceCreamOptions): void;
const options = { cherry: "yes" };
iceCream(options);
```

**Expected behavior:**

Error message recommending to add a type annotation to `options`.  This should be detectable because the source of the failing property is a `PropertyAssignment` in an object literal being assigned to a variable. (If there is no variable assignment, we could still recommend using ``"yes" as "yes"` although that isn't as pretty.)
Could also come with a codefix. Note that in JS the fix is slightly different: put `/** @type {IceCreamOptions} */` before the `const` or use `/** @type {"yes"} */ ("yes")`.

**Actual behavior:**

Error messages says `Type 'string' is not assignable to type '"yes" | "no"'.` with no information on how to fix it.

**Comment by Kingwl** [comment]
add "yes" as "yes" is soooooooooooooooooooooooo ugly  
are there any other way to declare the string literal type?

**Comment by RyanCavanaugh** [comment]
@Kingwl discussion at #10195

**Comment by Kingwl** [comment]
I'm going to improve this one,  but I found the type of diagnostic  is `DiagnosticMessage` or `DiagnosticMessageChain` that without position info, and I cannot get the un-assignable props in the object literal simply, 

As the problem，i'd like to replace `DiagnosticMessageChain` with `DiagnosticRelatedInformation` chain 
is `isRelatedTo` and that will hold the position info

ping @RyanCavanaugh  

Am I missing anything？
